api/features/views.py

```diff
@@ -108,6 +109,33 @@ def perform_create(self, serializer):
     def perform_update(self, serializer):
         serializer.save(project_id=self.kwargs.get("project_pk"))
 
+    def perform_destroy(self, instance):
+        self._create_feature_delete_audit_log(instance)
+        instance.delete()
+
+    def _create_feature_delete_audit_log(self, feature: Feature):
+        message = FEATURE_DELETED_MESSAGE % feature.name
+        project_audit_log = AuditLog(
+            author=self.request.user,
+            project=feature.project,
+            related_object_type=RelatedObjectType.FEATURE.name,
+            related_object_id=feature.id,
+            log=message,
+        )
+        audit_logs = [project_audit_log]
+        for environment in feature.project.environments.all():
```

Comment:
Hmm... rather than creating a log message for the feature in an environment (which doesn't really make sense), shouldn't we do something like the following: 

```python
for feature_state in feature.feature_states.filter(identity=None, feature_segment=None):
    audit_logs.append(AuditLog(..., related_object_type=RelatedObjectType.FEATURE_STATE.name, related_object_id=feature_state.id, ...))
audit_logs.append()
```

Reply:
We actually do use the message in  all the integrations:
```
"text": f"{log} by user {email}",

```

Reply:
Sorry, yes, I can see why my message wasn't clear. I wasn't suggesting that we remove the message, I was just suggesting here that we change the `related_object_type` and `related_object_id` to be that of the feature state, rather than the feature. I was being lazy and only highlighting the bits I was suggesting that we change.